src/dune_rules/lib.ml

```diff
@@ -1,6 +1,16 @@
 open! Dune_engine
 open! Import
-open Resolve.O
+open Resolve.Build.O
+
+module Resolve = struct
```

Comment:
IMO, things would be simpler without this re-mapping.

Reply:
and clearer especially, since `Resolve.return` would mean the same things in different files.